homeassistant/components/sensor/nsw_fuel_station.py

```diff
@@ -0,0 +1,120 @@
+"""
+Component to display the current fuel prices at a NSW fuel station.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/nsw_fuel_station/
+"""
+import datetime
+from typing import Optional
+import voluptuous as vol
+
+from homeassistant.components.light import PLATFORM_SCHEMA
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['nsw-fuel-api-client==1.0.7']
+
+CONF_STATION_ID = 'station_id'
+CONF_STATION_NAME = 'station_name'
+CONF_FUEL_TYPES = 'fuel_types'
+CONF_DEFAULT_FUEL_TYPES = ["E10", "U91", "E85", "P95", "P98", "DL",
+                           "PDL", "B20", "LPG", "CNG", "EV"]
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_STATION_ID): cv.positive_int,
+    vol.Required(CONF_STATION_NAME): cv.string,
+    vol.Optional(CONF_FUEL_TYPES, default=CONF_DEFAULT_FUEL_TYPES):
+        vol.All(cv.ensure_list, [cv.string]),
+})
+
+MIN_TIME_BETWEEN_UPDATES = datetime.timedelta(hours=1)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the NSW Fuel Station component."""
+    from nsw_fuel import FuelCheckClient
+
+    station_id = config[CONF_STATION_ID]
+    station_name = config[CONF_STATION_NAME]
+    fuel_types = config[CONF_FUEL_TYPES]
+
+    client = FuelCheckClient()
+    station_data = StationPriceData(client, station_id, station_name)
```

Comment:
The setup of the platform should fail if the station ID is wrong/not available and an error logged (log file and/or persistent notification). Otherwise will be users end up with a non-functional platform.

Reply:
Added persistent notification with error message:

<img width="504" alt="screen shot 2018-06-09 at 9 37 37 pm" src="https://user-images.githubusercontent.com/1289759/41191117-01fb7734-6c2e-11e8-8237-a53577b8dcf5.png">